src/modules/schedule/controller.ts

```diff
@@ -35,65 +33,40 @@ export async function removeSchedule(scheduleID: string): Promise<string> {
 
 
 // create a new schedule
-export async function createSchedule(created_by: string, term: string, schedule_name: string | undefined | null, is_public: boolean, class_IDs: string[] | undefined | null, primary_section_IDs: string[], secondary_section_IDs: string[]): Promise<Schedule> {
-  
-  const newID = new ObjectID()
-
-  const schedulePartsToCreate: minimumViableSchedule = { _id: newID, created_by: created_by, term: term }
-  if (schedule_name) {
-    schedulePartsToCreate.name = schedule_name
-  }
-
-  if (class_IDs) {
-    schedulePartsToCreate.class_IDs = class_IDs
-  }
+export async function createSchedule(created_by: string, term: TermInput, schedule_name: string | undefined | null, is_public: boolean | null | undefined, class_IDs: string[] | undefined | null, primary_section_IDs: string[] | undefined | null, secondary_section_IDs: string[] | undefined | null): Promise<Schedule> {
 
-  if (primary_section_IDs) {
-    schedulePartsToCreate.primary_section_IDs = primary_section_IDs
-  }
-
-  if (secondary_section_IDs) {
-    schedulePartsToCreate.secondary_section_IDs = secondary_section_IDs
-  }
-
-  schedulePartsToCreate.public = is_public
+  const schedulePartsToCreate: minimumViableSchedule = { created_by: created_by, term: term }
+  
+  schedulePartsToCreate.name = schedule_name ? schedule_name : undefined
+  schedulePartsToCreate.class_IDs = class_IDs ? class_IDs : undefined
+  schedulePartsToCreate.primary_section_IDs = primary_section_IDs ? primary_section_IDs : undefined
+  schedulePartsToCreate.secondary_section_IDs = secondary_section_IDs ? secondary_section_IDs : undefined
+  schedulePartsToCreate.is_public = is_public ? is_public : undefined
 
   const newSchedule = await ScheduleModel.create(schedulePartsToCreate)
```

Comment:
(continued from `resolver.ts` comment)

I know I suggested this, but I still think this is quite hard to work with. Consider the case where we add an extra field to the schedule schema: we would have to come here and refactor this code. 

I played around with how to more easily create (this also applies to update/edit), and passing the object directly into `create()` should work much cleaner:

```ts
// the `createSchedule()` function now takes in one argument `args`

const newSchedule = await ScheduleModel.create(args);
```

Frustratingly, if the graphql inputs can be null (no !), then users can explicitely pass in `null`. Mongoose will not use a default value if `null` is passed in, and so now there is a `null` value in a MongoDB document. When queried, mongoose will give a null value in a field declared non-null by graphql. Not too sure what the best fix for this is, but for now I just loop over the `args` argument and `delete` null fields.

Reply:
Regarding removing explicit null values, my input type is a bit complicated. Within it, I have another input type (the term) and an array of input types (the custom events). I was getting type issues with the return value of `omitBy` so I did some web searches and didn't see an easy solution. For now I adapted the code you initially shared on Discord.

Reply:
Hm yeah, seems like a recursive `omitBy` doesn't exist.